src/material/datepicker/multi-year-view.spec.ts

```diff
@@ -216,6 +216,34 @@ describe('MatMultiYearView', () => {
 
           expect(calendarInstance.date).toEqual(new Date(2017 + yearsPerPage * 2, JAN, 1));
         });
+
+        it('should go to the year that is focused', () => {
+          fixture.componentInstance.date = new Date(2017, MAR, 5);
+          fixture.detectChanges();
+          expect(calendarInstance.date).toEqual(new Date(2017, MAR, 5));
+
+          const year2022El = fixture.debugElement.nativeElement.querySelector(
+            '[data-mat-row="1"][data-mat-col="2"] button',
+          ) as HTMLElement;
+
+          dispatchFakeEvent(year2022El, 'focus');
```

Comment:
Would the test worked if you actually focused the element with `.focus()`?

Reply:
well, not sure exactly what the question is here. If we changed this line to `year2022El.focus()` then the test would fail when we check that the `focusSpy` was not called.

An alternative would be to actually focus the element with `.focus()`, but assert that the `focusSpy` was called exactly one.

I went with `dispathFakeEvent` because that's what we use in all the other tests, but I'm open to suggestions.